xbmc/pvr/PVRDatabase.cpp

```diff
@@ -289,19 +299,21 @@ bool CPVRDatabase::DeleteChannels()
   return DeleteValues("channels");
 }
 
-bool CPVRDatabase::Delete(const CPVRChannel& channel)
+bool CPVRDatabase::QueueDeleteQuery(const CPVRChannel& channel)
 {
   /* invalid channel */
   if (channel.ChannelID() <= 0)
     return false;
 
-  CLog::LogFC(LOGDEBUG, LOGPVR, "Deleting channel '{}' from the database", channel.ChannelName());
+  CLog::LogFC(LOGDEBUG, LOGPVR, "Queueing delete for channel '{}' from the database",
+              channel.ChannelName());
 
   Filter filter;
   filter.AppendWhere(PrepareSQL("idChannel = %u", channel.ChannelID()));
 
-  CSingleLock lock(m_critSection);
-  return DeleteValues("channels", filter);
+  std::string strQuery;
+  BuildSQL(PrepareSQL("DELETE FROM %s ", "channels"), filter, strQuery);
+  return CDatabase::QueueInsertQuery(strQuery);
```

Comment:
Question 1: Why the 'CDatabase::' prefix?
Question 2: Why 'QueueInsertQuery' and not 'QueueDeleteQuery'?

Reply:
2 is an autocomplete mistake, sorry.

It won't build without `CDatabase::` prefix. And I'm not sure why.

```
/Users/phunkyfish/kodi-repos/xbmc-origin/xbmc/pvr/PVRDatabase.cpp:316:27: error: no viable conversion from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char> >') to 'const PVR::CPVRChannel'
  return QueueDeleteQuery(strQuery);
                          ^~~~~~~~
/Users/phunkyfish/kodi-repos/xbmc-origin/xbmc/pvr/channels/PVRChannel.h:449:5: note: candidate constructor not viable: no known conversion from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char> >') to 'const PVR::CPVRChannel &' for 1st argument
    CPVRChannel(const CPVRChannel& tag) = delete;
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/string:875:5: note: candidate function
    operator __self_view() const _NOEXCEPT { return __self_view(data(), size()); }
    ^
/Users/phunkyfish/kodi-repos/xbmc-origin/xbmc/pvr/PVRDatabase.cpp:302:56: note: passing argument to parameter 'channel' here
bool CPVRDatabase::QueueDeleteQuery(const CPVRChannel& channel)
                                                       ^
1 error generated.
```

Reply:
If I change the name of `CPVRDatabase::QueueDeleteQuery` to `CPVRDatabase::QueueDelete` I no longer need the prefix.

Reply:
Hmm. Why does the compiler not pick the right QueueDeleteQuery? You pass a string, and we have an overload with a const string& why does it not just pick that?

Reply:
If we cannot solve that, I would stick with the CPVRDatabase prefix.

Reply:
Ya, it makes no sense at all. I don't know why.

Reply:
Okay.